cvat/apps/engine/cache.py

```diff
@@ -43,17 +44,30 @@ def __init__(self, dimension=DimensionType.DIM_2D):
         self._cache = caches['media']
 
     def _get_or_set_cache_item(self, key, create_function):
-        slogger.glob.info(f'Starting to get chunk from cache: key {key}')
-        item = self._cache.get(key)
-        slogger.glob.info(f'Ending to get chunk from cache: key {key}, is_cached {bool(item)}')
-        if not item:
+        def create_item():
             slogger.glob.info(f'Starting to prepare chunk: key {key}')
             item = create_function()
             slogger.glob.info(f'Ending to prepare chunk: key {key}')
+
             if item[0]:
+                item = (item[0], item[1], zlib.crc32(item[0].getbuffer()))
                 self._cache.set(key, item)
 
-        return item
+            return item
+
+        slogger.glob.info(f'Starting to get chunk from cache: key {key}')
+        item = self._cache.get(key)
+        slogger.glob.info(f'Ending to get chunk from cache: key {key}, is_cached {bool(item)}')
+        if not item:
+            item = create_item()
+        else:
+            # compare checksum
+            item_checksum = item[2] if len(item) == 3 else None
+            if item_checksum != zlib.crc32(item[0].getbuffer()):
+                slogger.glob.info(f'Recreating cache item {key} due to checksum mismatch')
+                item = create_item()
```

Comment:
I would propose to recreate chunk and compute checksum if checksum does not exist

Like:
```
if not item or len(item) != 3:
    item = create_item()
```